Check redirects introduced by docs changes

ptgott opened the pull request.

ptgott committed:
Check redirects introduced by docs changes

Configure the `Lint (Docs)` workflow to ensure that docs changes that
delete or rename a docs page also introduce a redirect.

ptgott commented:
Here's an example of a failed workflow run: https://github.com/gravitational/teleport/actions/runs/13294296457/job/37122360213

This resulted from the following commit: https://github.com/gravitational/teleport/pull/48546/commits/ff05d31e495233cf2d8ebbc242a02edc0416c9bb

fheinecke commented on .github/workflows/doc-tests.yaml:
```diff
@@ -53,6 +53,27 @@ jobs:
+          go-version: 'stable'
+
+      - name: Ensure docs changes include redirects
+        run: cd .github/shared-workflows/bot && go run main.go -workflow=docpaths -token="${{ steps.generate_token.outputs.token }}" -teleport-path="${{ github.workspace }}/teleport" -reviewers="${{ secrets.reviewers }}"
```
```suggestion
        env:
          TOKEN: ${{ steps.generate_token.outputs.token }}
          REVIEWERS: ${{ secrets.reviewers }}
        run: cd .github/shared-workflows/bot && go run main.go -workflow=docpaths -token="${TOKEN}" -teleport-path="${GITHUB_WORKSPACE}/teleport" -reviewers="${REVIEWERS}"
```

This should probably use env vars instead of interpolation with GHA contexts. If the context vars contain a double quote or other shell value, then the resulting command will not be what you expect.

ptgott replied:
Thanks!

fheinecke commented on .github/workflows/doc-tests.yaml:
```diff
@@ -53,6 +53,27 @@ jobs:
+        uses: actions/checkout@v4
+        with:
+          repository: gravitational/shared-workflows
+          path: .github/shared-workflows
```
I would recommend not putting this under `.github`. Github likes to introduce breaking changes without notification, and I wouldn't be surprised if they start treating this as a "special" directory at some point. Maybe `${{ runner.temp }}/shared-workflows`, or just `shared-workflows`?

ptgott committed:
Respond to fheinecke feedback

- Edit the checkout path for shared-workflows
- Use environment variables instead of interpolation with GHA contexts
